### PR TITLE
Allow renaming in autoautosummary

### PR DESCRIPTION
**Pull request by drasmuss**
This allows the module of classes/functions to be changed when using `automodule` or `autoautosummary`, so that things can be documented in different namespaces.

Note: depends on https://github.com/nengo/nengo-bones/pull/86

### CODE REVIEW

File: nengo_sphinx_theme/ext/autoautosummary.py [hunk 1]
```diff
@@ -122,12 +148,76 @@ def patch_autosummary_import_by_name():
 
     def import_by_name(name, prefixes):
         # Filter out problematic prefixes
-        prefixes = [p for p in prefixes if p is None or not name.startswith(p)]
-        return orig_f(name, prefixes)
+        # TODO: are there any situations where we want prefixes?
```
Comment (drasmuss) [inline nengo_sphinx_theme/ext/autoautosummary.py]: Not sure if there is anything we want to do about this TODO. This works fine in this repo, nengo core, and nengo-dl, so we weren't relying on the prefix functionality in any of those cases. But I'm not sure if there's a more permanent way to verify that we aren't relying on that functionality. Only thing that occurred to me is requiring that ``name`` contain ".", but that's not a reliable solution.

Comment (hunse) [reply nengo_sphinx_theme/ext/autoautosummary.py]: If an import can't be found by name, we'll obviously get an error, so IMO that's just as good as making our own error e.g. if `name` doesn't contain ".". I would just change this to say e.g. "We currently do not allow prefixes to be used". Then in the future if we need this, we can add it back in.

**Review by hunse — APPROVED** [review]

**Comment by hunse** [comment]
This fixes #47.

File: nengo_sphinx_theme/ext/autoautosummary.py (hunk 1 above)
Comment (drasmuss) [reply nengo_sphinx_theme/ext/autoautosummary.py]: Yeah that sounds good

**Pull request merged**